integration/fuse/src/main/java/alluxio/fuse/file/FuseFileOutStream.java

```diff
@@ -59,48 +64,70 @@ public class FuseFileOutStream implements FuseFileStream {
    *
    * @param fileSystem the Alluxio file system
    * @param authPolicy the Authentication policy
+   * @param pathLocks the path locks
    * @param uri the alluxio uri
    * @param flags the fuse create/open flags
    * @param mode the filesystem mode, -1 if not set
-   * @param status the uri status
    * @return a {@link FuseFileInOrOutStream}
    */
   public static FuseFileOutStream create(FileSystem fileSystem, AuthPolicy authPolicy,
-      AlluxioURI uri, int flags, long mode, Optional<URIStatus> status) {
+      LockPool<String> pathLocks, AlluxioURI uri, int flags, long mode) {
     Preconditions.checkNotNull(fileSystem);
     Preconditions.checkNotNull(authPolicy);
     Preconditions.checkNotNull(uri);
-    Preconditions.checkNotNull(status);
-    if (mode == AlluxioFuseUtils.MODE_NOT_SET_VALUE && status.isPresent()) {
-      mode = status.get().getMode();
-    }
-    long fileLen = status.map(URIStatus::getLength).orElse(0L);
-    if (status.isPresent()) {
-      if (AlluxioFuseOpenUtils.containsTruncate(flags) || fileLen == 0) {
-        // support create file then open with truncate flag to write workload
-        // support create empty file then open for write/read_write workload
-        AlluxioFuseUtils.deletePath(fileSystem, uri);
-        fileLen = 0;
-        if (LOG.isDebugEnabled()) {
-          LOG.debug(String.format("Open path %s with flag 0x%x for overwriting. "
-              + "Alluxio deleted the old file and created a new file for writing", uri, flags));
+    // Make sure file is not being read/written by current FUSE
+    RWLockResource lockResource = AlluxioFuseUtils.lock(pathLocks, uri.toString(), LockMode.WRITE,
```

Comment:
It looks like this call is not that high frequent. A full path check should be fine.

But I am also good with the hash way.

Reply:
the worry is that the full path is too long and the string takes a lot of memory since every read/write will push the string into the hash map

Reply:
consistency comes with sacrifices of memory